modeltrans/forms.py

```diff
@@ -0,0 +1,360 @@
+import itertools
+
+from django import forms
+from django.utils.translation import gettext_lazy as _
+
+from .conf import get_available_languages, get_default_language
+from .fields import get_instance_field_value
+from .translator import get_i18n_field
+from .utils import build_localized_fieldname, get_language
+
+LANGUAGE_OPTIONS = ["browser", "fallback"]
+
+
+class TranslationModelFormOptions(forms.models.ModelFormOptions):
+    """Add the translation form options to the Meta options."""
+
+    def __init__(self, options=None):
+        super().__init__(options)
+        self.languages = getattr(options, "languages", ["browser"])
+        self.fallback_language = getattr(options, "fallback_language", None)
+
+
+class TranslationModelFormMetaClass(forms.models.ModelFormMetaclass):
+    def __new__(mcs, name, bases, attrs):
+        """
+        Include all translation fields for translatable fields declared in the form.
+
+        We use the standard ModelForm field declaration procedure of the ModelFormMetaClass,
+        so that all fields are declared during creation of the new object.
+
+        Different actions are required for different Meta class options:
+        1) fields - if fields are declared explicitly with the fields option, the related i18n fields will not be
+        included, and these have to be added to base_fields.
+        2) exclude - if fields are declared implicitly with the exclude option, the i18n translation fields, as well as
+        the i18n field itself, will be included. We will remove the i18n field.
+
+        Here all languages are included, so that in the form we can curate which fields are available according to
+        settings of the form instance (e.g. kwarg overrides, or model instance fallbacks).
+        """
+        new_class = super().__new__(mcs, name, bases, attrs)
+
+        # from ModelFormMetaClass, needed for calling fields for model
+        base_formfield_callback = None
+        for b in bases:
+            if hasattr(b, "Meta") and hasattr(b.Meta, "formfield_callback"):
+                base_formfield_callback = b.Meta.formfield_callback
+                break
+        formfield_callback = attrs.pop("formfield_callback", base_formfield_callback)
+
+        # get options, with extra translation form options
+        opts = new_class._meta = TranslationModelFormOptions(getattr(new_class, "Meta", None))
+        model_class = opts.model
+
+        # TODO CHECK:
+        # how does modeltrans add fields during exclude, normally? i.e. what determines the complete language list?
+        languages = get_available_languages()
```

Comment:
is "get_available_languages" hoe modeltrans het normaal doet? Dus dan is dit okee toch?